test/Glue/Integration/Application/KernelTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSourcingWorkshopTest\Glue\Integration\Application;
+
+use DateTimeImmutable;
+use EventSourcingWorkshop\EventSourcing\Domain\Aggregate\Aggregate;
+use EventSourcingWorkshop\EventSourcing\Infrastructure\Aggregate\EventStreamAggregateRepository;
+use EventSourcingWorkshop\EventSourcing\Infrastructure\Persistence\SQLiteEventStore;
+use EventSourcingWorkshop\EventSourcing\Infrastructure\Streaming\TraverseEventStreamAndSaveStatusInSqlite;
+use EventSourcingWorkshop\Glue\Application\Kernel;
+use PHPUnit\Framework\TestCase;
+
+/** @covers \EventSourcingWorkshop\Glue\Application\Kernel */
+final class KernelTest extends TestCase
+{
+    public function testKernelProvidesAllUtilities(): void
+    {
+        $kernel = new Kernel();
+
+        self::assertEquals(1, $kernel->db->fetchOne('SELECT 1'));
+        self::assertGreaterThanOrEqual(new DateTimeImmutable(), $kernel->clock->now());
+        self::assertInstanceOf(SQLiteEventStore::class, $kernel->eventStore);
+        self::assertInstanceOf(TraverseEventStreamAndSaveStatusInSqlite::class, $kernel->traverseEventStream);
+        self::assertInstanceOf(EventStreamAggregateRepository::class, $kernel->aggregateRepository(Aggregate::class));
+
+        $kernel->ensureMigrationsRan();
```

Comment:
Ugly, but I didn't want to over-complicate the kernel, for now.

Good enough to verify that it doesn't break with auto-dependency-upgrades